### PR TITLE
Properly handle invalid cvar replications without blocking netmessage

### PR DESCRIPTION
**Pull request by BobTheBob9**
Reimplements `ns_server_name` and `ns_server_desc` replication without causing issues on clients that do not have the change locally, and prevents bad replicated cvars from breaking client connect

### CODE REVIEW

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
Didn't work in testing.

In particular, with this PR I could join a random just fine but hosting a (Windows) dedi an having someone on `v1.12.0` join did not work. They got stuck in loading screen with no error message or anything in logs.

To reproduce:
1. Host dedi with this PR
2. Have someone on `v1.12.0` join said dedi server
3. They should now be stuck in eternal loading screen.

**EDIT:** Relevant sections from log from client that couldn't join. They had to kill Northstar via taskmanager

```
[23:32:36] [NORTHSTAR] [info] Requesting server list from https://northstar.tf
[23:32:36] [NORTHSTAR] [info] Got 144 servers
[23:32:38] [NORTHSTAR] [info] Attempting to register the local server to the master server.
[23:32:41] [SCRIPT UI] [info] trying to authenticate with server Temp Testing server with password 
[23:32:42] [NORTHSTAR] [info] Attempting authentication with server of id "d1af1dec8e680450ced0c08edf6f9782"
[23:32:42] [SCRIPT UI] [info] Showing user info for UID 0 on hardware 
[23:32:42] [SCRIPT UI] [info] Showing user info for UID 0 on hardware 
[23:32:42] [SCRIPT UI] [info] Showing user info for UID 0 on hardware 
[23:32:42] [SCRIPT UI] [info] UICodeCallback_LevelShutdown: mp_lobby
[23:32:42] [SCRIPT UI] [info] Hiding all matchmaking elems due to UpdateMatchmakingStatus thread ending
[23:32:42] [SCRIPT UI] [info] menu_ServerBrowserMenu menu closed
[23:32:42] [SCRIPT UI] [info] menu_LobbyMenu menu closed
[23:32:42] [SCRIPT UI] [info] menu_MainMenu menu opened
[23:32:42] [SCRIPT UI] [info] =========== Show: MainMenuPanel ===========
[23:32:42] [SCRIPT UI] [info] SAVEGAME try is valid savegame
[23:32:42] [SCRIPT UI] [info] SAVEGAME is valid  savegame
[23:32:42] [NORTHSTAR] [info] DestroyVM CLIENT 0x2035401e400
[23:32:42] [SCRIPT UI] [info] UICodeCallback_LevelLoadingStarted: 
[23:32:42] [NORTHSTAR] [info] HostState: GameShutdown
[23:32:42] [NORTHSTAR] [info] Player eRaid_oPlaiN disconnected: "Game Shutting Down"
[23:32:42] [NORTHSTAR] [info] Player eRaid_oPlaiN disconnected: "Game Shutting Down"
[23:32:42] [NORTHSTAR] [warning] attempted to write pdata of size 0!
[23:32:42] [NATIVE SV] [info] CVScriptGameSystem::LevelShutdownPreEntity()
[23:32:42] [NORTHSTAR] [info] DestroyVM SERVER 0x2034a3d4600
[23:32:43] [RPAK_FSYS] [info] LoadPakAsync mp_forwardbase_kodai_loadscreen.rpak 209
[23:32:43] [RPAK_FSYS] [info] LoadStreamPak: pc_all.starpak
[23:32:43] [SCRIPT UI] [info] UICodeCallback_UpdateLoadingLevelName: mp_forwardbase_kodai
[23:32:43] [FILESYSTM] [info] MountVPK vpk/client_mp_forwardbase_kodai.bsp
[23:32:43] [NORTHSTAR] [info] Set playlist to tdm
[23:32:43] [NORTHSTAR] [info] 
R2
Map: mp_forwardbase_kodai
Players: 2 (0 bots) / 16 humans
Build: 6968
Server Number: 2

[23:32:43] [SCRIPT UI] [info] UICodeCallback_UpdateLoadingLevelName: mp_forwardbase_kodai
[23:32:43] [RPAK_FSYS] [info] LoadPakAsync mp_forwardbase_kodai.rpak 210
[23:32:43] [RPAK_FSYS] [info] LoadStreamPak: pc_all.starpak
[23:32:43] [RPAK_FSYS] [info] LoadStreamPak: pc_all(04).starpak
[23:32:43] [RPAK_FSYS] [info] LoadStreamPak: pc_all(06).starpak
[23:32:47] [NORTHSTAR] [error] ExploitFixes.cpp: NET_SetConVar (client): Blocked dangerous/invalid msg: Invalid flags (0x4), var is ns_server_desc
[23:32:48] [NORTHSTAR] [error] Failed reading masterserver response: got fastify error response
[23:32:48] [NORTHSTAR] [error] {"error":{"enum":"NO_GAMESERVER_RESPONSE","msg":"Couldn't reach game server: failed to connect to auth port: request timed out"},"request_id":"cfdeh193ngoqhtcdte2g","success":false}

[23:33:55] [NORTHSTAR] [info] HostState: GameShutdown
[23:33:55] [NORTHSTAR] [warning] ServerAuthenticationManager::StopPlayerAuthServer was called while m_bRunningPlayerAuthThread is false
[23:33:55] [NATIVE SV] [info] CVScriptGameSystem::LevelShutdownPreEntity()
[23:33:55] [NORTHSTAR] [info] Set playlist to tdm
[23:33:55] [SCRIPT UI] [info] UICodeCallback_LevelLoadingFinished:  (false)
[23:33:55] [SCRIPT UI] [info] UICodeCallback_ActivateMenus: menu_MainMenu
[23:33:55] [SCRIPT UI] [info] UICodeCallback_ActivateMenus: menu_MainMenu
[23:33:55] [NORTHSTAR] [info] DestroyVM UI 0x2034abc0300
```


Also, going back to `v1.12.0` on dedi and retrying client joining worked just fine so issue seems to be with PR.

**Comment by BobTheBob9** [comment]
re-requesting review as previous testing didn't properly test pr

**Comment by GeckoEidechse** [comment]
Will take another look once I'm back in a few days. If anyone else is able to do it earlier, feel free to do so and leave a review. ^^

Either way with this merged we'd probably have to version gate again to prevent breakage. so not sure when the best time to merge/release it is.

**Comment by GeckoEidechse** [comment]
(closing was miss-click, sorry >.<)

**Comment by GeckoEidechse** [comment]
Uh what was the best way to test this again? ^^"

**Review by ASpoonPlaysGames — APPROVED** [review]
Code looks good, Gecko's testing should still also be valid I think?

**Comment by Alystrasz** [comment]
Dunno if this can be merged, but it can't be updated to main because of branch protection rules.

**Comment by GeckoEidechse** [comment]
> Dunno if this can be merged, but it can't be updated to main because of branch protection rules.

Thanks for the heads-up. I got enough perms so I updated it now ^^

**Comment by GeckoEidechse** [comment]
Also sorta issue with this PR is that it's incompatible with previous versions so it requires a version gate.

**Review by catornot — APPROVED** [review]
Was running this branch with a friend for a week or so with no issues :)

hopefully this will get merged soon.

File: primedev/shared/exploit_fixes/exploitfixes.cpp [hunk 1]
```diff
@@ -120,19 +120,30 @@ bool, __fastcall, (void* pMsg)) // 48 8B D1 48 8B 49 18 48 8B 01 48 FF 60 10
...
+				if (pVar)
+				{
+					memcpy(
+						entry->name,
+						pVar->m_ConCommandBase.m_pszName,
+						strlen(pVar->m_ConCommandBase.m_pszName) + 1); // Force name to match case
+
+					if (!pVar->IsFlagSet(FCVAR_REPLICATED))
+					{
+						return BLOCKED_INFO(
```
Comment (BobTheBob9) [inline primedev/shared/exploit_fixes/exploitfixes.cpp:138]: unsure when exactly this was made to return but it shouldn't! with this return in place, i believe this pr does not fix the issue it is attempting to fix

Comment (BobTheBob9) [reply primedev/shared/exploit_fixes/exploitfixes.cpp:138]: in my original change this was just an spdlog::warn without a return, not return BLOCKED_INFO

Comment (catornot) [reply primedev/shared/exploit_fixes/exploitfixes.cpp:138]: here is a fix for that https://github.com/R2Northstar/NorthstarLauncher/pull/683

Comment (GeckoEidechse) [reply primedev/shared/exploit_fixes/exploitfixes.cpp]: Fixed via e334f49f28c70fa9a309de9d7dbf0f9e7008aaad

**Review by GeckoEidechse — APPROVED** [review]
Retested rn and no breaking change to older servers which is awesome :D
No clue why it broke when I tested before.

Code review was done by others earlier before already.

**Pull request merged**